packages/in_app_purchase/lib/src/in_app_purchase/app_store_connection.dart

```diff
@@ -124,11 +124,15 @@ class AppStoreConnection implements InAppPurchaseConnection {
   @override
   Future<PurchaseVerificationData> refreshPurchaseVerificationData() async {
     await SKRequestMaker().startRefreshReceiptRequest();
-    String receipt = await SKReceiptManager.retrieveReceiptData();
-    return PurchaseVerificationData(
-        localVerificationData: receipt,
-        serverVerificationData: receipt,
-        source: IAPSource.AppStore);
+    try {
+      String receipt = await SKReceiptManager.retrieveReceiptData();
+      return PurchaseVerificationData(
+          localVerificationData: receipt,
+          serverVerificationData: receipt,
+          source: IAPSource.AppStore);
+    } catch (e) {
+      return null;
```

Comment:
I wonder if we should print out a warning message about the error?

Reply:
Done